src/utils/wallet.js

```diff
@@ -314,19 +351,35 @@ class Wallet {
         this.accounts[accountId] = true
     }
 
-    async addAccessKey(accountId, contractId, publicKey) {
-        try {
-            return await this.getAccount(accountId).addKey(
-                publicKey,
-                contractId,
-                '', // methodName
-                ACCESS_KEY_FUNDING_AMOUNT
-            )
-        } catch (e) {
-            if (e.type === 'AddKeyAlreadyExists') {
-                return true;
+
+    /********************************
+    recovering a second account attempts to call this method with the currently logged in account and not the tempKeyStore 
+    ********************************/
+    async addAccessKey(accountId, contractId, publicKey, fullAccess = false) {
+        const { account, has2fa } = await this.getAccountAndState()
+        if (!contractId) {
+            contractId = account.accountId
+        }
+        if (has2fa) {
+            return await this.twoFactor.addKey(account, publicKey, contractId, fullAccess)
+        } else {
+            try {
+                if (fullAccess) {
+                    return await this.getAccount(accountId).addKey(publicKey)
+                } else {
+                    return await this.getAccount(accountId).addKey(
+                        publicKey,
+                        contractId,
+                        '', // methodName
+                        ACCESS_KEY_FUNDING_AMOUNT
+                    )
+                }
+            } catch (e) {
+                if (e.type === 'AddKeyAlreadyExists') {
```

Comment:
what will happen if twoFactor.addKey encounters this error? Are errors propagated?

Reply:
Looking into this.